src/Module/Profile/Attachment/Upload.php

```diff
@@ -0,0 +1,164 @@
+<?php
+/**
+ * @copyright Copyright (C) 2010-2022, the Friendica project
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace Friendica\Module\Profile\Attachment;
+
+use Friendica\App;
+use Friendica\Core\Config\Capability\IManageConfigValues;
+use Friendica\Core\L10n;
+use Friendica\Core\Session\Capability\IHandleUserSessions;
+use Friendica\Database\Database;
+use Friendica\Model\Attach;
+use Friendica\Model\User;
+use Friendica\Module\Response;
+use Friendica\Navigation\SystemMessages;
+use Friendica\Network\HTTPException\InternalServerErrorException;
+use Friendica\Util\Profiler;
+use Friendica\Util\Strings;
+use Psr\Log\LoggerInterface;
+
+class Upload extends \Friendica\BaseModule
+{
+	/** @var Database */
+	private $database;
+
+	/** @var IHandleUserSessions */
+	private $userSession;
+
+	/** @var IManageConfigValues */
+	private $config;
+
+	/** @var SystemMessages */
+	private $systemMessages;
+
+	/** @var bool */
+	private $isJson;
+
+	public function __construct(SystemMessages $systemMessages, IManageConfigValues $config, IHandleUserSessions $userSession, Database $database, L10n $l10n, App\BaseURL $baseUrl, App\Arguments $args, LoggerInterface $logger, Profiler $profiler, Response $response, array $server, array $parameters = [])
+	{
+		parent::__construct($l10n, $baseUrl, $args, $logger, $profiler, $response, $server, $parameters);
+
+		$this->database       = $database;
+		$this->userSession    = $userSession;
+		$this->config         = $config;
+		$this->systemMessages = $systemMessages;
+	}
+
+	protected function post(array $request = [])
+	{
+		if ($this->isJson = !empty($request['response']) && $request['response'] == 'json') {
+			$this->response->setType(Response::TYPE_JSON, 'application/json');
+		}
+
+		$nick  = $this->parameters['nickname'];
+		$owner = User::getOwnerDataByNick($nick);
+		if (!$this->database->isResult($owner)) {
```

Comment:
!empty() would be better, I thought this isResult is a legacy function?

Reply:
Absolutely, it only makes sense with the direct return of the database query method since it counts the rows of an eventual result.